test/libsolidity/syntaxTests/viewPureChecker/inline_assembly_instructions_disallowed_pure.sol

```diff
@@ -0,0 +1,84 @@
+contract C {
+    function f() public pure {
+        assembly {
+            pop(sload(0))
+            sstore(0, 1)
+            pop(gas())
+            pop(address())
+            pop(balance(0))
+            pop(selfbalance())
+            pop(caller())
+            pop(callvalue())
+            pop(extcodesize(0))
+            extcodecopy(0, 1, 2, 3)
+            pop(returndatasize())
+            returndatacopy(0, 1, 2)
+            pop(extcodehash(0))
+            pop(create(0, 1, 2))
+            pop(create2(0, 1, 2, 3))
+            pop(call(0, 1, 2, 3, 4, 5, 6))
+            pop(callcode(0, 1, 2, 3, 4, 5, 6))
+            pop(delegatecall(0, 1, 2, 3, 4, 5))
+            pop(staticcall(0, 1, 2, 3, 4, 5))
+            selfdestruct(0)
+            log0(0, 1)
+            log1(0, 1, 2)
+            log2(0, 1, 2, 3)
+            log3(0, 1, 2, 3, 4)
+            log4(0, 1, 2, 3, 4, 5)
+            pop(chainid())
+            pop(basefee())
+            pop(origin())
+            pop(gasprice())
+            pop(blockhash(0))
+            pop(coinbase())
+            pop(timestamp())
+            pop(number())
+            pop(difficulty())
+            pop(gaslimit())
+
+            // These two are disallowed too but the error suppresses other errors.
+            //pop(msize())
+            //pop(pc())
+        }
+    }
+}
+// ====
+// EVMVersion: >=london
+// ----
+// Warning 5740: (742-1153): Unreachable code.
+// TypeError 2527: (79-87): Function declared as pure, but this expression (potentially) reads from the environment or state and thus requires "view".
+// TypeError 8961: (101-113): Function cannot be declared as pure because this expression (potentially) modifies the state.
```

Comment:
Ah sorry! Somehow I missed this.

Reply:
Yeah. For each visibility level there are two files: one with allowed and one with disallowed opcodes :)